### PR TITLE
[TSK-88] implement boards crud

### PR DESCRIPTION
**Pull request by kostya4201337**

### CODE REVIEW

**Comment by adsukharev** [comment]
@Hoocki you review his PRs first. After you approve it, I will review it.

File: client/src/pages/workspace/Workspace.tsx
Comment (Hoocki) [inline client/src/pages/workspace/Workspace.tsx]: Right now there're predefined values in the useState. In the future, you are gonna send a request to the server to get these boards. To avoid making many changes in the future, let's make it extensible first.

1. Create a new folder `services` in the root
2. Create a file `board.ts`
3. create a function `getBoardsService()` which for now return your mock boards. In the future we will implement here requests to the server
4. Go back to Workspace component
5. Set initial values to empty array `useState<BoardEntity[]>([]);`
6. Create a function `getBoards()`. Inside call `getBoardsService()` and set new boards to your state
7. Now we need to somehow call `getBoards()` automatically when we open a page. Create useEffect hook https://react.dev/reference/react/useEffect and call getBoards() inside. UseEffect is used to automatically trigger some actions depending on some conditions/events.
```
useEffect(() => {
        getBoards();
    }, []);
```
`, []` second parameter tells react when rerun useEffect. `[]` - means run it on startup. You can do this way `[boards]` it means rerun useEffect whene boards are changed.
Have a look more about useEffect and try to test it with diffrent conditions.
Same shit will be for other cases where you expect get data from the server

File: client/src/model/entity/CardEntity.tsx
Comment (Hoocki) [inline client/src/model/entity/CardEntity.tsx]: ICardEntity

File: client/src/model/entity/BoardEntity.tsx
Comment (Hoocki) [inline client/src/model/entity/BoardEntity.tsx]: rename to IBoardEntity

File: client/src/pages/workspace/Workspace.tsx
Comment (Hoocki) [inline client/src/pages/workspace/Workspace.tsx]: maybe you can remove `<div>` and use style in `<List>`
`<List style={{display: 'inline-flex', verticalAlign: 'middle'}}>`

File: client/src/pages/workspace/board/AddBoard.tsx
Comment (Hoocki) [inline client/src/pages/workspace/board/AddBoard.tsx]: add type for `AddBoard`

File: client/src/pages/workspace/board/AddBoard.tsx
Comment (Hoocki) [inline client/src/pages/workspace/board/AddBoard.tsx]: style in styles.css

File: client/src/pages/workspace/board/board.tsx
Comment (Hoocki) [inline client/src/pages/workspace/board/board.tsx]: add type for `Board`

File: client/src/pages/workspace/board/mutableInput.component.tsx
Comment (Hoocki) [inline client/src/pages/workspace/board/mutableInput.component.tsx]: create `props.ts` and put this code there

File: client/src/pages/workspace/board/mutableInput.component.tsx
Comment (Hoocki) [inline client/src/pages/workspace/board/mutableInput.component.tsx]: bad name for function

File: client/src/pages/workspace/board/AddBoard.tsx
Comment (Hoocki) [inline client/src/pages/workspace/board/AddBoard.tsx:1]: create folder `addBoard` and put this file (include css and props which refer to `addBoard`)

File: client/src/pages/workspace/board/board.tsx
Comment (Hoocki) [inline client/src/pages/workspace/board/board.tsx]: instead of create 2 useState, create 1 with type `IBoardCreation`

File: client/src/pages/workspace/board/props.tsx
Comment (Hoocki) [inline client/src/pages/workspace/board/props.tsx]: create props.ts in `board/addBoard` and put code there

File: client/src/pages/workspace/board/style.css
Comment (Hoocki) [inline client/src/pages/workspace/board/style.css]: remove space

File: client/src/pages/workspace/board/mutableInput.style.css
Comment (Hoocki) [inline client/src/pages/workspace/board/mutableInput.style.css]: space

File: client/src/pages/workspace/board/mutableInput.component.tsx
Comment (Hoocki) [inline client/src/pages/workspace/board/mutableInput.component.tsx:1]: bad name for file

File: client/src/pages/workspace/Workspace.tsx
Comment (konstantinwas) [reply client/src/pages/workspace/Workspace.tsx]: vertical align doesnt work anyway, i just forgot to delete it

File: client/src/pages/workspace/board/AddBoard.tsx
Comment (kostya4201337) [reply client/src/pages/workspace/board/AddBoard.tsx:30]: i dont know why, but when i put it in the style.css it doesnt work

File: client/src/model/dto/IBoardCreation.tsx
Comment (Hoocki) [inline client/src/model/dto/IBoardCreation.tsx:1]: for models and props type of file should be `.ts`

File: client/src/pages/workspace/Workspace.tsx
Comment (Hoocki) [inline client/src/pages/workspace/Workspace.tsx]: you can remove type for `fetchedBoards`, because it's obviously we expect `IBoardEntity[]`

File: client/src/pages/workspace/Workspace.tsx
Comment (Hoocki) [inline client/src/pages/workspace/Workspace.tsx]: you should call your `addBoardService` to get the new board and use it in `setBoards`

File: client/src/pages/workspace/board/addBoard/AddBoard.tsx
Comment (Hoocki) [inline client/src/pages/workspace/board/addBoard/AddBoard.tsx]: why are you create new board here, when you have `addBoardService`

File: client/src/pages/workspace/board/AddBoard.tsx
Comment (Hoocki) [reply client/src/pages/workspace/board/AddBoard.tsx:1]: move folder `addBoard` to `pages/workspace`, because you use it in workspace not in board

File: client/src/pages/workspace/board/mutableInput/MutableInput.component.tsx
Comment (Hoocki) [inline client/src/pages/workspace/board/mutableInput/MutableInput.component.tsx]: name of this method doesn't tell what it is doing

File: client/src/pages/workspace/board/addBoard/AddBoard.tsx
Comment (Hoocki) [inline client/src/pages/workspace/board/addBoard/AddBoard.tsx]: it would be better if you renamed the props and functions to explicit names. 
For example: `onCreate` is better to replace with `addBoard` and so on.

**Review by Hoocki — APPROVED** [review]

File: client/src/pages/workspace/board/addBoard/AddBoard.tsx
Comment (konstantinwas) [inline client/src/pages/workspace/board/addBoard/AddBoard.tsx]: i didnt put it in the style.css file because TextField overrides it

Comment (adsukharev) [reply client/src/pages/workspace/board/addBoard/AddBoard.tsx]: Have you tried margin prop? https://mui.com/material-ui/react-text-field/#margin
what is your unit of measurement? px, %, rem, em, vh ? 

All in all, you don't have to have styles in your code. Fix it everywhere

File: package.json
Comment (adsukharev) [inline package.json:1]: this pckage.json is outside of the project

File: client/src/model/entity/IBoardEntity.ts
Comment (adsukharev) [inline client/src/model/entity/IBoardEntity.ts:1]: entity files are on the backend. In your case, just `IBoard`.
Then restructure your models bu following this style https://github.com/Hoocki/Trello/tree/master/client/src/models

**Comment by adsukharev** [comment]
@Hoocki just a reminder if you are revewing a PR, you have to run the app and check manually how it works. Except when PR is really small

File: client/src/pages/workspace/Workspace.tsx
Comment (adsukharev) [inline client/src/pages/workspace/Workspace.tsx:15]: Why do you need  here empty array?

File: client/src/pages/workspace/Workspace.tsx
Comment (adsukharev) [inline client/src/pages/workspace/Workspace.tsx]: move it yo styles.css

File: client/src/pages/workspace/addBoard/AddBoard.tsx
Comment (adsukharev) [inline client/src/pages/workspace/addBoard/AddBoard.tsx]: 1) to styles.css.
2) when you use numbers, your web site becomes static and inresponsive when you change size of window. After opening your web page, try to change browser window to small size, you will see that your components are out of the window. But they should be become smaller as well. https://elementor.com/help/whats-the-difference-between-px-em-rem-vw-and-vh/

File: client/src/pages/workspace/addBoard/AddBoard.tsx
Comment (adsukharev) [inline client/src/pages/workspace/addBoard/AddBoard.tsx]: styles and don't use px

File: client/src/pages/workspace/board/board.tsx
Comment (adsukharev) [inline client/src/pages/workspace/board/board.tsx]: it is so ugly to see styles in the code

File: client/src/pages/workspace/mutableInput/MutableInput.component.tsx
Comment (adsukharev) [inline client/src/pages/workspace/mutableInput/MutableInput.component.tsx]: move it to the dedicated method

File: client/src/pages/workspace/addBoard/AddBoard.tsx
Comment (adsukharev) [inline client/src/pages/workspace/addBoard/AddBoard.tsx]: are you trying to save space?

File: client/src/pages/workspace/mutableInput/MutableInput.component.tsx
Comment (adsukharev) [inline client/src/pages/workspace/mutableInput/MutableInput.component.tsx]: name of the file and name of the component are different

File: client/src/pages/workspace/mutableInput/MutableInput.style.css
Comment (adsukharev) [inline client/src/pages/workspace/mutableInput/MutableInput.style.css:1]: styles.css

File: client/src/pages/workspace/mutableInput/MutableInput.component.tsx
Comment (adsukharev) [inline client/src/pages/workspace/mutableInput/MutableInput.component.tsx:1]: this component is not unique for particular component.
You need to create a new folder `client/src/components/mutableInput` and move you files there

File: client/src/pages/workspace/board/style.css
Comment (adsukharev) [inline client/src/pages/workspace/board/style.css:1]: styles

File: client/src/pages/workspace/board/board.tsx
Comment (adsukharev) [inline client/src/pages/workspace/board/board.tsx:1]: name component files from Uppercase. `Board.tsx`

File: client/src/pages/workspace/board/board.tsx
Comment (adsukharev) [inline client/src/pages/workspace/board/board.tsx]: Why cannot you add className directly to MutableInput? Your divs look redundant

File: client/src/pages/workspace/board/board.tsx
Comment (adsukharev) [inline client/src/pages/workspace/board/board.tsx]: If you start using grid component from material ui https://mui.com/material-ui/react-grid/ or flexbox https://developer.mozilla.org/ru/docs/Learn/CSS/CSS_layout/Flexbox
 https://mui.com/system/flexbox/ your design will be more responsive

**Review by adsukharev — COMMENTED** [review]
don't resolve comments to see history of the comments

File: client/src/pages/workspace/mutableInput/MutableInput.component.tsx
Comment (adsukharev) [inline client/src/pages/workspace/mutableInput/MutableInput.component.tsx]:
```suggestion
    const [edit, setEdit] = useState<boolean>(false);
```
However in general, it's a bad name because it's a verb but type is boolean

File: client/src/pages/workspace/mutableInput/MutableInput.component.tsx
Comment (adsukharev) [inline client/src/pages/workspace/mutableInput/MutableInput.component.tsx]: ` onKeyPress` The attribute is marked as deprecated . Why cannot you use onChange?

File: client/src/pages/workspace/addBoard/AddBoard.tsx
Comment (kostya4201337) [reply client/src/pages/workspace/addBoard/AddBoard.tsx:76]: Button overrides it, when i put it in the css it doesnt work

Comment (adsukharev) [reply client/src/pages/workspace/addBoard/AddBoard.tsx]: In that case you still can separate styles and code.
Create a new file, let's say customStyles.ts and put there your styles:
```
export const addButtonStyle = {
    borderRadius: 20,
    backgroundColor: "#ffffff",
    fontSize: "18px",
    borderLeft: "2px #575757 solid",
    borderTop: "2px #575757 solid",
};
```

Then use it in your component:
```
<Button
                    style={addButtonStyle}
```

File: client/src/pages/workspace/Workspace.tsx
Comment (konstantinwas) [reply client/src/pages/workspace/Workspace.tsx:15]: because changes doesnt show without it?

File: client/src/pages/workspace/mutableInput/MutableInput.component.tsx
Comment (konstantinwas) [reply client/src/pages/workspace/mutableInput/MutableInput.component.tsx]: ?

Comment (adsukharev) [reply client/src/pages/workspace/mutableInput/MutableInput.component.tsx]: File name : MutableInput.component
Component name : MutableInput

File: client/src/pages/workspace/Workspace.tsx
Comment (adsukharev) [reply client/src/pages/workspace/Workspace.tsx:15]: Yes, but what does empty array mean?

File: client/src/pages/workspace/addBoard/AddBoard.tsx
Comment (adsukharev) [inline client/src/pages/workspace/addBoard/AddBoard.tsx]: Extract it and reuse in both textfields

File: client/src/pages/workspace/Workspace.tsx
Comment (adsukharev) [inline client/src/pages/workspace/Workspace.tsx]: Styles

File: client/src/pages/workspace/mutableInput/MutableInput.component.tsx
Comment (adsukharev) [reply client/src/pages/workspace/mutableInput/MutableInput.component.tsx]: You fixed only in this case. Make the same change in other places

File: client/src/mutableInput/MutableInput.component.tsx
Comment (adsukharev) [inline client/src/mutableInput/MutableInput.component.tsx]: Why do you add id evrywhere?

File: client/src/mutableInput/MutableInput.component.tsx
Comment (adsukharev) [inline client/src/mutableInput/MutableInput.component.tsx]: Move this condition to the selectField(). This method should be responsible for validation

File: client/src/pages/workspace/mutableInput/MutableInput.component.tsx
Comment (adsukharev) [reply client/src/pages/workspace/mutableInput/MutableInput.component.tsx]: You sill did not fix it

File: client/src/pages/workspace/mutableInput/MutableInput.component.tsx
Comment (adsukharev) [reply client/src/pages/workspace/mutableInput/MutableInput.component.tsx:1]: add a folder `components`

File: client/src/mutableInput/styles.css
Comment (adsukharev) [inline client/src/mutableInput/styles.css]: You use hyphens in styles above, but here camelCase. Choose one style and follow it everywhere

File: client/src/pages/workspace/addBoard/AddBoard.tsx
Comment (adsukharev) [inline client/src/pages/workspace/addBoard/AddBoard.tsx]: why do you need this wrapper div if inside only one component?

**Review by adsukharev — COMMENTED** [review]
You design is stil not responsive. If you create several boards or just reduce side of the window you will see it
Screenshot 2023-11-03 at 10 14 18

**Comment by adsukharev** [comment]
@Hoocki Have you really run his app and manually tested it? Deletion functioanlly does not work correctly

File: client/src/pages/workspace/Workspace.tsx
Comment (konstantinwas) [reply client/src/pages/workspace/Workspace.tsx:15]: run on startup

**Comment by kostya4201337** [comment]
> You design is stil not responsive. If you create several boards or just reduce side of the window you will see it Screenshot 2023-11-03 at 10 14 18

shouldn't it be like that? it doesn't scale in the notion too
image

File: client/src/pages/workspace/addBoard/AddBoard.tsx
Comment (adsukharev) [inline client/src/pages/workspace/addBoard/AddBoard.tsx]: don't create variables in the component. You should create it in a state or outside of the component. The problem is that your variable will be recreated for each rerender operation. You can check it if you write console.log(textFieldInputStyle) and do some actions

File: client/src/component/mutableInput/props.ts
Comment (adsukharev) [inline client/src/component/mutableInput/props.ts]:
```suggestion
    setText: (value: string) => void;
```

File: client/src/pages/workspace/addBoard/AddBoard.tsx
Comment (adsukharev) [inline client/src/pages/workspace/addBoard/AddBoard.tsx]: why do you add styles here and not to className?

**Comment by adsukharev** [comment]
> @Hoocki Have you really run his app and manually tested it? Deletion functioanlly does not work correctly

@Hoocki ?

@adsukharev no, i just checked only code

File: client/src/pages/workspace/addBoard/AddBoard.tsx
Comment (adsukharev) [inline client/src/pages/workspace/addBoard/AddBoard.tsx:32]: I mean in general, instead of inputProps why not to use className?

**Review by adsukharev — COMMENTED** [review]
I want to wait for @Hoocki to answer on my question. Then I will approve

**Comment by kostya4201337** [comment]
> I want to wait for @Hoocki to answer on my question. Then I will approve

he did

File: client/src/pages/workspace/addBoard/AddBoard.tsx
Comment (konstantinwas) [reply client/src/pages/workspace/addBoard/AddBoard.tsx:32]: i dont know how to make it in css

**Comment by adsukharev** [comment]
> > I want to wait for @Hoocki to answer on my question. Then I will approve
> 
> he did

File: client/src/component/mutableInput/mutableInput.tsx
Comment (adsukharev) [inline client/src/component/mutableInput/mutableInput.tsx:1]: MutableInput.tsx

File: client/src/pages/workspace/board/props.ts
Comment (adsukharev) [inline client/src/pages/workspace/board/props.ts:3]:
```suggestion
export type BoardProps = {
```

**Review by adsukharev — APPROVED** [review]

**Comment by kostya4201337** [comment]
uhhhhhhhhh

**Comment by kostya4201337** [comment]
i thought you aproved it without comments, it appeared after i merged

**Pull request merged**
